packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -639,94 +557,10 @@ async function fixed({ file, args = {}, reporter, cache }) {
   }
 }
 
-async function notMemoizedtraceSVG({ file, args, fileArgs, reporter }) {
-  const potrace = require(`potrace`)
-  const svgToMiniDataURI = require(`mini-svg-data-uri`)
-  const trace = promisify(potrace.trace)
-  const defaultArgs = {
-    color: `lightgray`,
-    optTolerance: 0.4,
-    turdSize: 100,
-    turnPolicy: potrace.Potrace.TURNPOLICY_MAJORITY,
-  }
-  const optionsSVG = _.defaults(args, defaultArgs)
-  const options = healOptions(pluginOptions, fileArgs, file.extension)
-  let pipeline
-  try {
-    pipeline = sharp(file.absolutePath).rotate()
-  } catch (err) {
-    reportError(`Failed to process image ${file.absolutePath}`, err, reporter)
-    return null
-  }
-
-  pipeline
-    .resize(options.width, options.height, {
-      position: options.cropFocus,
-    })
-    .png({
-      compressionLevel: options.pngCompressionLevel,
-      adaptiveFiltering: false,
-      force: args.toFormat === `png`,
-    })
-    .jpeg({
-      quality: options.quality,
-      progressive: options.jpegProgressive,
-      force: args.toFormat === `jpg`,
-    })
-
-  // grayscale
-  if (options.grayscale) {
-    pipeline = pipeline.grayscale()
-  }
-
-  // rotate
-  if (options.rotate && options.rotate !== 0) {
-    pipeline = pipeline.rotate(options.rotate)
-  }
-
-  // duotone
-  if (options.duotone) {
-    pipeline = await duotone(
-      options.duotone,
-      args.toFormat || file.extension,
-      pipeline
-    )
-  }
-
-  const tmpDir = require(`os`).tmpdir()
-  const tmpFilePath = `${tmpDir}/${file.internal.contentDigest}-${
-    file.name
-  }-${crypto
-    .createHash(`md5`)
-    .update(JSON.stringify(fileArgs))
-    .digest(`hex`)}.${file.extension}`
-
-  await new Promise(resolve =>
-    pipeline.toFile(tmpFilePath, (err, info) => {
-      resolve()
-    })
-  )
```

Comment:
generation of temporary input file was refactored into separate function `notMemoizedPrepareTraceSVGInputFile` and added memoization on top that use `tmpFilePath` as key